src/Sentry.Unity/ScreenshotAttachment.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.IO;
+using Sentry;
+using Sentry.Extensibility;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+
+namespace Sentry.Unity
+{
+    internal class ScreenshotAttachment : Attachment
+    {
+        public ScreenshotAttachment(IAttachmentContent content)
+            : base(AttachmentType.Default, content, "screenshot.jpg", "image/jpeg") { }
+    }
+
+    internal class ScreenshotAttachmentContent : IAttachmentContent
+    {
+        private readonly SentryMonoBehaviour _behaviour;
+        private readonly SentryOptions _options;
+
+        public ScreenshotAttachmentContent(SentryOptions options, SentryMonoBehaviour behaviour)
+        {
+            _behaviour = behaviour;
+            _options = options;
+        }
+
+        public Stream GetStream()
+        {
+            if (!_behaviour.MainThreadData.IsMainThread())
+            {
+                _options.DiagnosticLogger?.LogDebug("Won't capture screenshot because we're not on the main thread");
+                return new MemoryStream();
+            }
+            else
+            {
+                // The async version - uses a custom stream & captures in a coroutine
+                // return new ScreenshotCaptureStream(_options, _behaviour);
+
+                // Captures current screenshot synchronously
+                try
+                {
+                    var texture = ScreenCapture.CaptureScreenshotAsTexture();
+                    var bytes = texture.EncodeToJPG();
+                    _options.DiagnosticLogger?.LogDebug("Screenshot captured: {0} bytes", bytes.Length);
+                    return new MemoryStream(bytes);
+                }
+                catch (Exception ex)
+                {
+                    _options.DiagnosticLogger?.LogError("Couldn't capture screenshot", ex);
+                }
+            }
+            return new MemoryStream();
+        }
+    }
+
+    // The async version, currently unused because we want the screenshot to be as early as possible after the error
```

Comment:
OK it's quite a bit of code so maybe we store this in the GitHub issue as a comment so we don't lose it. but here it's a lot of deadcode